lib/utils.rb

```diff
@@ -56,3 +56,33 @@ def exit_unless_dir(path)
   end
   return true
 end
+
+def expand_node_ranges(nodes)
+  new_nodes = []
+  nodes.each do |node|
+    if node.match(/.*\[[0-9]+.*[0-9]+\]$/)
+      m = node.match(/^(.*)\[(.*)\]$/)
+      prefix = m[1]
+      suffix = m[2]
+      ranges = suffix.split(',')
+      ranges.each do |range|
+        if range.match(/-/)
+          num_1, num_2 = range.split('-')
+          padding = num_1.match(/^0+/)
```

Comment:
I think this fails for `node00-01`:

```
["node000", "node001"]
```

(Only spotted/checked for this cos @WilliamMcCumstie and I noticed something similar in a method he's derived from https://github.com/DougEverly/nodeattr/blob/master/lib/nodeattr.rb)

Reply:
Ok thanks Mark!

My method is partially derived from the same place (along with something I wrote for adminware that was never added - https://github.com/alces-software/adminware/pull/73/commits/f0913f1d04e3fd2dd49852657e55f624969375b2).
I switched methods a couple of times while writing this but I landed on the conclusion that the method in `nodeattr.rb` is flawed - its handling of comma-separated ranges differs from nodeattr's

Can I just clarify what your input values were - was it node[00-01]?

Reply:
Yup `node[00-01]` yields 2-zero padded results, i.e. 000 and 001. Similarly `node[0-1]` would yield 1-zero padded results i.e. 00 and 01.